### PR TITLE
Remove xamarin-jdk

### PR DESCRIPTION
**Pull request by gtmgianni**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] The commit message includes the cask’s name and version.

### CODE REVIEW

File: Casks/xamarin-jdk.rb [hunk 1]
```diff
@@ -1,10 +1,15 @@
 cask 'xamarin-jdk' do
-  version '8.101'
-  sha256 '680de8ddead3867fc34e7ff380f437c7ddb8dc75eb606186a3e8ae7e3b8c7fbc'
+  version '8.172.b11'
+  sha256 '80212f7748c2671b89b6085000717c747851db004409dce3bb97f3a2aeb91cdd'
 
-  url "https://download.xamarin.com/Installer/MonoForAndroid/jdk-#{version.major}u#{version.minor}-macosx-x64.dmg"
+  # download.oracle.com/otn-pub/java/jdk was verified as official when first introduced to the cask
+  url "http://download.oracle.com/otn-pub/java/jdk/#{version.major}u#{version.minor}-#{version.patch}/a58eab1ec242421181065cdc37240b08/jdk-#{version.major}u#{version.minor}-macosx-x64.dmg",
```
Comment (commitay) [inline Casks/xamarin-jdk.rb]: This is the same URL as the `java8` Cask, where did it come from?

Comment (gtmgianni) [reply Casks/xamarin-jdk.rb]: https://docs.microsoft.com/en-us/xamarin/android/troubleshooting/questions/update-jdk?tabs=vsmac

Comment (gtmgianni) [reply Casks/xamarin-jdk.rb]: @commitay Also, for some reason `#{version.patch}` isn't being substituted in the `url` block, and I'm unsure why that is.

Comment (commitay) [reply Casks/xamarin-jdk.rb]: The Xamarin/Microsoft instructions are to install the Oracle JDK so we can remove this Cask as a duplicate of `java8`

**Comment by commitay** [comment]
Commenting here so it isn't hidden by GitHub.

The current upstream instructions point to the Oracle download page to download Java, making this Cask a duplicate of `java8`

https://docs.microsoft.com/en-us/xamarin/android/troubleshooting/questions/update-jdk?tabs=vsmac

**Pull request merged**